packages/kit/src/runtime/server/page/load_node.js

```diff
@@ -98,9 +98,32 @@ export async function load_node({
 				let response;
 
 				if (/^[a-zA-Z]+:/.test(url)) {
-					// external fetch
-					const request = new Request(url, /** @type {RequestInit} */ (opts));
-					response = await options.hooks.serverFetch.call(null, request);
+					// possibly external fetch
+					if (typeof request.host !== 'undefined') {
+						const { hostname: fetchHostname } = new URL(url);
+						const [serverHostname] = request.host.split(':');
+
+						// allow cookie passthrough for "same-origin"
```

Comment:
maybe we should also go ahead and handle the `include` case as well?

Reply:
To clarify, if the user has specified `credentials: 'include'` I believe that we should also send cookies in that case? https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters

Reply:
We can't deal with cross-origin cookies, e.g. fetch from `domain.com` to `otherthing.com` would normally send cookies for `otherthing.com` to `otherthing.com`, but this never goes through `domain.com`

For same-origin targets it's treated the same as `credentials: same-origin`, though, since we're just checking that the fetch isn't `credentials: omit`.